reports/vaibhav.md

```diff
@@ -0,0 +1,174 @@
+      then continue execution.
+   3. Users can specify the bear they want to debug using `--debug-bears`.
+   4. Users can specify to debug bears using a `.coafile`.
+
+2. A new command `settings` is included in coala's Debugging Interface to
+   inspect Bear settings in the debugging environment. It displays all passed
+   settings of a Bear and their values so bear developer can quickly inspect
+   them.
+
+3. Profiler Interface added in coala.
+   1. The profiler will start by profiling the `run()` method of bears because
+      this is the part where bear writers will spend time on, as rest of the
+      part like loading the files, collecting the settings, etc. are done by
+      coala itself.
+   2. `--profile-bears` or `profile_bears` (using .coafile) is the main argument
+      to enable profiling.
+   3. Accept an additional parameter `directory_path` through which bear
+      developers can specify where to store the profiled data files.
+
+#### Challenges
+
+I had difficulties with testing of Debugger interface. With the help of `Mock`,
+test for debugger interface was created, though code coverage was 100% but
+`mock` test was not the ideal test, so after hours of discussion with Mischa we
+decided to capture the `pdb stdout` and assert the the output with the real
+output but problem arised, the ideal tests passes but somehow it didn't showed
+the code coverage to 100% not only that it also removed the coverage of the
+previous tests, at first I thought created test have some problem but seems
+like `pytest-cov` had a problem, so to confirm that I created a
+[test repo](https://github.com/RaiVaibhav/test_repo) and enable the
+[CI](https://travis-ci.org/RaiVaibhav/test_repo) on that repo to check the
+coverage report, it also showed same failure. So, I created a issue on
+`pytest-cov` repo but actual problem was, both coverage and debugger use the
+same trace feature, so it was not possible to run coverage and debugger in
+parallel or say it is not possible to chain the trace functions, so Mischa gave
+a idea to restore the trace callbacks and then run test, it worked finally so I
+included both ideal and `Mock` test so that the coverage remain 100, but in all
+of these I lagged around 3 week from my schedule. So I couldn't implement one of
+the proposed feature.
+
+#### Work to be done
+
+A tool which will provide the ability to inspect result instances like
+`origin`, `diffs`, `file`, `severity`, `message`, `aspects` and several other
+attributes in debugging environment or in a seperate python console where all
+where user can easily access there values. Information about result will help
+bear developer in creating a better performant Bears.
+
+Result inspection tool will not only help in debugging the bear results but
+also help in testing the bear by asserting the various attributes of expected
+result with actual result.
+
+So, after `GSoC` my first task will be to add the inspection tool in the coala
```

Comment:
no ... you have zero responsibility after GSoC.

it is up to you if you do this or not.

if you dont, someone else will.  You've created the framework.

Reply:
So I will have to add a issue link here ?